### PR TITLE
[DEV-2230] Include observation table operation into Context API Object

### PR DESCRIPTION
**Pull request by yanhuisg**
## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/api/view.py [hunk 1]
```diff
@@ -1687,10 +1686,13 @@ def create_observation_table(
         ...   context_id=context_id,
         ... )
         """
-        pruned_graph, mapped_node = self.extract_pruned_graph_and_node()
+        # pylint: disable=import-outside-toplevel
+        from featurebyte.api.context import Context
```
Comment (yanhuisg) [inline featurebyte/api/view.py:1690]: to avoid circular import in Context api object

File: featurebyte/api/source_table.py [hunk 2]
```diff
@@ -1016,6 +1015,7 @@ def create_observation_table(
         ... )
         """
         # pylint: disable=import-outside-toplevel
+        from featurebyte.api.context import Context
```
Comment (yanhuisg) [inline featurebyte/api/source_table.py:1018]: to avoid circular import in Context api object

File: featurebyte/api/use_case_or_context_mixin.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        observation_table_name: str
+            New observation table to be added.
+
+        Examples
+        --------
+        >>> use_case = catalog.get_use_case("use_case")
+        >>> use_case.add_observation_table(observation_table_name)  # doctest: +SKIP
+        """
+        observation_table = ObservationTable.get(observation_table_name)
+        observation_table.update(update_payload={"context_id": self.id}, allow_update_local=False)
```
Comment (jevonyeoh) [inline featurebyte/api/use_case_or_context_mixin.py:69]: how do we know that the `self.id` corresponds to a context? this operation feels a little dangerous as it feels like we could set any id into this `context_id` field.

Comment (yanhuisg) [reply featurebyte/api/use_case_or_context_mixin.py:69]: I will move it into the Context api object

**Review by jevonyeoh — APPROVED** [review]

**Pull request merged**